services/database/user.ts

```diff
@@ -35,4 +40,100 @@ async function getUsers() {
     return users;
 }
 
-export { getUser, getUsers };
+/* 
+TODO (6/27/21): https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#create 
+doc does not specify what happens if the create operation fails - does that mean it should always
+succeed? in that case, do we need to return anything or do we just return response 200 every time?
+*/
+
+/**
+ * Creates a new user corresponding to newUserData and returns the newly created user
+ * @param newTeacherData - data corresponding to the new teacher user
+ * @returns - the newly created teacher user
+ */
+
+async function createTeacher(
+    newTeacherData: CreateTeacherInput,
+    userId: number,
+): Promise<Teacher> {
+    const teacher = await prisma.teacher.create({
+        data: {
+            ...newTeacherData,
+            users: {
+                connect: { id: userId },
```

Comment:
Is this connect required? Can you not just specify id: userId in `data`?

Reply:
Don't think so, the `prisma.teacher.create` function does not have a `user_id` field, but according to the prisma documentation, the `connect` field is how you link a new record to an existing record: https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#connect-an-existing-record

Reply:
The `id` field of teacher is actually the `user.id` as specified by the foreign key constraint on the `id` field in teacher

Reply:
Could you try passing in just the userId as `id` instead of using connect to see if it works?